update design of enrollment and order marts

KatelynGit opened the pull request:
### What are the relevant tickets?
https://github.com/mitodl/hq/issues/4296

### Description (What does it do?)
Changed order of enrollment mart to alphabetical for readability. Moved order columns to orders mart from enrollment mart it makes more sense there design wise. Improved definition.

### How can this be tested?
dbt build --select marts__combined__orders
dbt build --select marts__combined_course_enrollment_detail

rachellougee commented on src/ol_dbt/models/marts/combined/marts__combined__orders.sql:
```diff
@@ -103,6 +108,11 @@ with bootcamps__ecommerce_order as (
+        , null as order_tax_rate
+        , null as order_tax_rate_name
+        , null as order_tax_amount
+        , null as order_total_price_paid_plus_tax
```
Per the updated description `For other platform, this is the same as order_total_price_paid`, this shouldn't be null, right?

KatelynGit replied:
I've changed this tp pull the order_total_price_paid for the other platforms should be fixed now

rachellougee reviewed (commented):
For re-arranging the field order in marts__combined_course_enrollment_detail, I think we should also consider the primary key for the table, not just sort alphabetically? These coupon-related fields are now placed first, but these are not primary keys for the table and most are NULL. I don't have a strong opinion but from a user standpoint I would expect `platform` and `courserunenrollment_id` to be the first columns.

Just a question, why don't we sort `marts__combined__orders` alphabetically for consistency?

KatelynGit commented:
> For re-arranging the field order in marts__combined_course_enrollment_detail, I think we should also consider the primary key for the table, not just sort alphabetically? These coupon-related fields are now placed first, but these are not primary keys for the table and most are NULL. I don't have a strong opinion but from a user standpoint I would expect `platform` and `courserunenrollment_id` to be the first columns.
> 
> Just a question, why don't we sort `marts__combined__orders` alphabetically for consistency?

I've also changed the order mart so it's alphabetical order.  But I've also changed both to add a few key joining columns at the top.
